components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -860,9 +869,19 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
                 throw err;
             }
 
+            let projectEnvVarsPromise: Promise<ProjectEnvVar[]> = Promise.resolve([]);
```

Comment:
IMO it's the 3rd time I see this code. Could it be extracted? Would also eliminated the "noisy" let-if combination.